src/Xhgui/Util.php

```diff
@@ -20,4 +23,137 @@ public static function simpleUrl($url)
         return preg_replace('/\=\d+/', '', $url);
     }
 
+    /**
+     * @return string
+     */
+    public static function getXHProfFileName()
+    {
+        if (empty($_SERVER['REQUEST_TIME_FLOAT'])) {
+            $t = explode('.', microtime(true));
+        } else {
+            $t = explode('.', $_SERVER['REQUEST_TIME_FLOAT']);
+        }
+        // join float part to main part and pad it to make every filename same length
+        return $t[1].str_pad($t[2], 4, 0, STR_PAD_RIGHT).'.data.xhprof';
+    }
+
+    /**
+     * Serialize data for storage
+     *
+     * @param      $data
+     * @param bool $profiles
+     *
+     * @return false|string
+     */
+    public static function getDataForStorage($data, $profiles = true)
+    {
+        if ($profiles) {
+            $serializer = Xhgui_Config::read('save.handler.serializer', 'json');
+        } else {
+            $serializer = Xhgui_Config::read('save.handler.meta_serializer', 'php');
+        }
+
+        switch ($serializer) {
+            case 'json':
+                return json_encode($data);
+                break;
+            case 'serialize':
```

Comment:
I don't think we should be using serialized at all. It is far too risky to have build an application that deserializes data obtained from the network.

Reply:
As far as I remember original FB code uses serialization instead of json. Having this makes GUI compatible with FB code.

Reply:
This project has never aimed to be compatible with the original Facebook data though.